File: nos2jdbc-tutorial-spring/src/main/java/nos2jdbc/tutorial/spring/entity/nonauto/MemberYm.java
package nos2jdbc.tutorial.spring.entity.nonauto;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import nos2jdbc.annotation.NoFk;
import nos2jdbc.annotation.NonAuto;

@Entity @NonAuto
public class MemberYm {
    @Id
    public Integer y;
    @Id
    public Integer m;
    
    @ManyToOne @NoFk
    public MemberKey member;
    
    @OneToMany(mappedBy = "ym")
    public List<MemberItem> itemList;
    

}

File: nos2jdbc-tutorial-kotlin-spring/src/main/kotlin/nos2jdbc/tutorial/kotlinspring/entity/nonauto/MemberYm.kt
package nos2jdbc.tutorial.kotlinspring.entity.nonauto

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany

import nos2jdbc.annotation.NoFk
import nos2jdbc.annotation.NonAuto

@Entity @NonAuto
class MemberYm {
    @Id
    var y: Int? = null
    @Id
    var m: Int? = null
    
    @ManyToOne @NoFk
    var member: MemberKey? = null
    
    @OneToMany(mappedBy = "ym")
    var itemList: List<MemberItem> = mutableListOf()
    

}
